api/src/wfl/module/covid.clj

```diff
@@ -1,34 +1,23 @@
 (ns wfl.module.covid
   "Manage the Sarscov2IlluminaFull pipeline."
-  (:require [clojure.data.json :as json]
+  (:require [clojure.edn :as edn]
             [clojure.set :as set]
-            [clojure.string :as str]
-            [clojure.tools.logging.readable :as log]
             [wfl.api.workloads :as workloads :refer [defoverload]]
             [wfl.jdbc :as jdbc]
             [wfl.module.batch :as batch]
-            [wfl.service.datarepo :as datarepo]
-            [wfl.service.firecloud :as firecloud]
-            [wfl.service.google.storage :as gcs]
             [wfl.service.postgres :as postgres]
             [wfl.service.rawls :as rawls]
             [wfl.util :as util]
             [wfl.wfl :as wfl])
-  (:import [java.time OffsetDateTime]))
+  (:import [java.time OffsetDateTime]
+           [java.util UUID]))
 
 (def pipeline "Sarscov2IlluminaFull")
 
-;; TODO: implement COVID workload creation
-;;  - make sure permissions/inputs are right upfront
-;;  - dispatch on source/sink/executor
-;;  - store information into database
-(defn create-covid-workload!
-  [tx {:keys [source sink executor] :as request}])
-
 (defn ^:private get-snapshots-from-workspace
   [workspace])
 
-(defn start-covid-workload!
+(defn start-covid-workload
```

Comment:
Not sure... It was originally when the function could throw or has some side effect but pretty much most of what we do involves has effects so I've fallen out with it.

Reply:
I agree they are overused.  A database read should not be `!`.

Reply:
It's a `start!`!
One thing to take not of, start should start the source too so it can write the initial last_checked time to be the `:started` time